src/PieChart/PieArcSeries/PieArcIcon.tsx

```diff
@@ -0,0 +1,71 @@
+import React from 'react';
+import chroma from 'chroma-js';
+
+import type { ArcData } from './PieArcSeries';
+
+interface InjectedProps {
+  data: ArcData;
+  centroid: (data: ArcData) => [number, number];
+  color: string;
+}
+
+interface OwnProps {
+  iconSize?: number;
+  renderIcon?: (params: {
+    data: ArcData['data'];
+    color: string;
+  }) => React.ReactElement | null;
+}
+
+type Props = InjectedProps & OwnProps;
+
+export type PieArcIconProps = OwnProps & Partial<InjectedProps>;
+
+const defaultRenderIcon: OwnProps['renderIcon'] = ({ data }) => {
```

Comment:
Not sure I follow what this is for

Reply:
I'll rename this thing to `formatter` and will stop passing icon component through `data.metadata`